### PR TITLE
Add nil checks for Azure CSI translation

### LINKED ISSUE #78524: Need nil checks for Azure CSI translation checks
**What happened**:
Some of the Azure CSI translation checks in `TranslateInTreePVToCSI` and `TranslateInTreeInlineVolumeToCSI` needs to perform NPE checks for pointer based fields as pointed out in https://github.com/kubernetes/kubernetes/pull/77703#pullrequestreview-243797085 and https://github.com/kubernetes/kubernetes/pull/77703#discussion_r288992067

**What you expected to happen**:
NPE checks should be there

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.15
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
As #78524 stated, we should check pointer against nil before dereferencing.

**Which issue(s) this PR fixes**:
Fixes #78524

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @ddebroy @liggitt

File: staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go
Comment (davidz627) [inline staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go]: @andyzhangx is this something we want to do?

File: staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go
Comment (davidz627) [inline staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go]: to preserve same behavior I think we still want to check for `len==0` case and not fill in an empty string in the `VolumeAttribute`

File: staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go
Comment (davidz627) [inline staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go]: what if this is `nil`

**Comment by tedyu** [comment]
@davidz627 
Can you take another look ?

**Comment by davidz627** [comment]
/lgtm
@andyzhangx lgtm before I approve as I'm not sure if some of these fields being `nil` is an error instead of just omission from the translation

File: staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go
Comment (ddebroy) [inline staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go]: Slightly unrelated to the specific nil checks (which looks lgtm), Q for @andyzhangx: why do we need FSType passed twice - as a field in CSI source as well as a volume attribute. If not needed, can one of them be removed?

**Review by ddebroy — COMMENTED** [review]
One q below. Otherwise lgtm.

File: staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go
Comment (andyzhangx) [reply staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go]: @ddebroy we support fstype in both CSI and driver parameter settings, this PR lgtm, thanks.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Comment by tedyu** [comment]
@davidz627 
Please let me know if there is something else I need to do for this PR.

**Pull request merged**
